src/utils/applySearchAndFilter.ts

```diff
@@ -0,0 +1,29 @@
+import { Person } from '../types';
+import { filterPeople } from './filterPeopleBy';
+import { sortPeople } from './sortPeopleBy';
+
+export const applySearchAndFilter = (
+  people: Person[],
+  searchParams: URLSearchParams,
+): Person[] => {
+  const stringParams = searchParams.toString();
+
+  if (!stringParams) {
+    return people;
+  }
+
+  let currentPeople = people;
+
+  if (stringParams.includes('centuries')
+    || stringParams.includes('sex=')
+    || stringParams.includes('query')
+  ) {
```

Comment:
You may want to put these search and sort options into an enum and use it throughout your code, like in filterPeopleBy, PepopleFilter, etc. Won't look like magic then.